be/src/storage/tablet_meta_manager.cpp

```diff
@@ -1294,10 +1294,76 @@ Status TabletMetaManager::remove(DataDir* store, TTabletId tablet_id) {
         if (!clear_pending_rowset(store, &batch, tablet_id).ok()) {
             LOG(WARNING) << "clear rowset add to batch failed";
         }
+        if (!clear_persistent_index(store, &batch, tablet_id).ok()) {
+            LOG(WARNING) << "clear persistent index add to batch failed";
+        }
     }
     return meta->write_batch(&batch);
 }
 
+Status TabletMetaManager::remove_table_meta(DataDir* store, TTableId table_id) {
+    KVStore* meta = store->get_meta();
+    WriteBatch batch;
+    bool is_primary = false;
+    std::vector<int64_t> delete_tablet;
+    rocksdb::ColumnFamilyHandle* cf = store->get_meta()->handle(META_COLUMN_FAMILY_INDEX);
+    auto traverse_tabletmeta_func = [&](std::string_view key, std::string_view value) -> bool {
+        TabletMetaPB tablet_meta_pb;
+        bool parsed = tablet_meta_pb.ParseFromArray(value.data(), value.size());
+        if (!parsed) {
+            LOG(WARNING) << "bad tablet meta pb data, tablet_meta key: " << key;
+        } else {
+            is_primary = tablet_meta_pb.schema().keys_type() == KeysType::PRIMARY_KEYS;
+            if (tablet_meta_pb.table_id() == table_id) {
+                auto st = batch.Delete(cf, string(key));
+                if (!st.ok()) {
+                    LOG(WARNING) << "batch.Delete failed, key:" << key;
+                } else if (is_primary) {
+                    if (!clear_log(store, &batch, tablet_meta_pb.tablet_id()).ok()) {
+                        LOG(WARNING) << "clear_log add to batch failed";
+                    }
+                    if (!clear_del_vector(store, &batch, tablet_meta_pb.tablet_id()).ok()) {
+                        LOG(WARNING) << "clear delvec add to batch failed";
+                    }
+                    if (!clear_rowset(store, &batch, tablet_meta_pb.tablet_id()).ok()) {
+                        LOG(WARNING) << "clear rowset add to batch failed";
+                    }
+                    if (!clear_pending_rowset(store, &batch, tablet_meta_pb.tablet_id()).ok()) {
+                        LOG(WARNING) << "clear rowset add to batch failed";
+                    }
+                    if (!clear_persistent_index(store, &batch, tablet_meta_pb.tablet_id()).ok()) {
+                        LOG(WARNING) << "clear persistent index add to batch failed";
+                    }
```

Comment:
is there a remove tablet code we could reuse?  for example, extract a common method

Reply:
> is there a remove tablet code we could reuse? for example, extract a common method

done